### PR TITLE
Add Code tab to Number Input page

### PR DESCRIPTION
**Pull request by Lisa159**

### CODE REVIEW

**Review by raquelarrojo — CHANGES_REQUESTED** [review]

File: website/pages/components/number-input/index.tsx [hunk 1]
```diff
@@ -1,21 +1,21 @@
 import Head from "next/head";
 import type { ReactElement } from "react";
-import NumberInputUsagePage from "../../../screens/components/number-input/usage/NumberInputUsagePage";
+import NumberInputCodePage from "../../../screens/components/number-input/code/NumberInputCodePage";
 import NumberInputPageLayout from "../../../screens/components/number-input/NumberInputPageLayout";
 
-const Usage = () => {
+const Code = () => {
   return (
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/pages/components/number-input/index.tsx:7]: Done

File: website/screens/components/number-input/code/examples/basic.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  DxcStack,
+} from "@dxc-technology/halstack-react";
+
+const code = `() => {
+  return (
+    <DxcInset space="large">
+      <DxcStack gutter="large">
+        <DxcNumberInput label="Default" />
+      </DxcStack>
+    </DxcInset>
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/examples/basic.ts]: Done

File: website/screens/components/number-input/code/NumberInputCodePage.tsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        title: "Uncontrolled",
+        content: (
+          <>
+            <Example example={uncontrolled} defaultIsVisible />
+          </>
+        ),
+      },
+      {
+        title: "Error Handling",
+        content: (
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/NumberInputCodePage.tsx:230]: Done

File: website/screens/components/number-input/code/examples/basic.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/examples/basic.ts]: Done

File: website/screens/components/number-input/code/NumberInputCodePage.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        </tr>
+      </DxcTable>
+    ),
+  },
+  {
+    title: "Examples",
+    subSections: [
+      {
+        title: "Basic Usage",
+        content: (
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/NumberInputCodePage.tsx:206]: Done

File: website/screens/components/number-input/code/examples/controlled.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  };
+  const onBlur = ({ value }) => {
+    setValue(value);
+  };
+
+  return (
+    <DxcInset space="large">
+      <DxcNumberInput
+        label="Default"
+        value={value}
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/examples/controlled.ts:18]: Done

File: website/screens/components/number-input/code/examples/uncontrolled.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    console.log(input.value);
+  };
+
+  return (
+    <DxcInset space="large">
+      <DxcStack gutter="large" align="start">
+        <DxcNumberInput
+          label="Uncontrolled"
+          defaultValue="1"
+          ref={inputRef}
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/examples/uncontrolled.ts:22]: Done

File: website/screens/components/number-input/code/examples/error.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        <DxcNumberInput
+          label="Custom range error"
+          value={value}
+          onChange={onChange}
+          error={errorMessage == undefined ? "" : "Invalid number."}
+          min={5}
+          max={20}
+        />
+      </DxcStack>
+    </DxcInset>
```
[the remark this answers is not in the document]
Comment (Lisa159) [reply website/screens/components/number-input/code/examples/error.ts]: Done

File: website/screens/components/number-input/code/examples/errorHandling.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+  return (
+    <DxcInset space="large">
+      <DxcNumberInput
+        label="Custom range error"
+        value={value}
+        onChange={onChange}
+        onBlur={onBlur}
+        error={errorMessage == undefined ? "" : errorMessage}
+        min={5}
```
Comment (raquelarrojo) [inline website/screens/components/number-input/code/examples/errorHandling.ts:24]: As this check is done in the onChange and onBlur, here we can keep it as: `error={errorMessage}`.

**Review by raquelarrojo — CHANGES_REQUESTED** [review]

**Review by raquelarrojo — APPROVED** [review]

**Pull request merged**